torture/src/supervisor/workload.rs

```diff
@@ -470,14 +489,43 @@ impl Workload {
         Ok(())
     }
 
+    async fn exercise_rollback(&mut self, rr: &comms::RequestResponse) -> anyhow::Result<()> {
+        // TODO: n_blocks should also depend on the max rollback supported by NOMT.
+        let n_snapshots = self.state.committed.len();
+        let n_blocks_to_rollback = std::cmp::min(
+            self.state.rng.gen_range(1..self.max_rollback_blocks) as usize,
+            n_snapshots - 1,
+        );
+
+        let last_sync_seqn = self.state.last_snapshot().sync_seqn;
+        self.state
+            .committed
+            .truncate(n_snapshots - n_blocks_to_rollback);
+        // The application of a rollback counts as increased sync_seq.
+        self.state.committed.last_mut().unwrap().sync_seqn = last_sync_seqn + 1;
+
+        if n_blocks_to_rollback == 0 {
+            trace!("No available blocks to perform rollback with");
+            return Ok(());
+        }
+
+        trace!("exercising rollback of {} blocks", n_blocks_to_rollback);
+        rr.send_request(crate::message::ToAgent::Rollback(n_blocks_to_rollback))
+            .await?;
+
+        self.ensure_snapshot_validity(rr).await?;
+        Ok(())
+    }
+
     async fn spawn_new_agent(&mut self) -> anyhow::Result<()> {
         assert!(self.agent.is_none());
         controller::spawn_agent_into(&mut self.agent).await?;
         let workdir = self.workdir.path().display().to_string();
+        let rollback = self.state.biases.rollback > 0.0;
```

Comment:
FWIW, this is fine for now, but in the future we want some workloads to be running with rollback enabled and some with rollback disabled.